modules/dex/src/lib.rs

```diff
@@ -494,69 +559,159 @@ pub mod module {
+							(min_contribution_a, min_contribution_b),
+							(target_provision_a, target_provision_b),
+						)
+					} else {
+						(
+							(min_contribution_b, min_contribution_a),
+							(target_provision_b, target_provision_a),
+						)
+					};
+					TradingPairStatuses::<T>::insert(
+						trading_pair,
+						TradingPairStatus::Provisioning(ProvisioningParameters {
+							min_contribution,
+							target_provision,
+							accumulated_provision: provisioning_parameters.accumulated_provision,
+							not_before,
+						}),
+					);
+				}
+				_ => return Err(Error::<T>::MustBeProvisioning.into()),
+			}
+
 			Ok(().into())
 		}
 
-		#[pallet::weight((<T as Config>::WeightInfo::disable_trading_pair(), DispatchClass::Operational))]
+		/// Enable a new trading pair
+		/// if status is `Disabled`, enable the trading pair directly,
+		/// if status is `Provisioning`, enable it after calculate the initial lp token amount.
+		#[pallet::weight((<T as Config>::WeightInfo::enable_trading_pair(), DispatchClass::Operational))]
 		#[transactional]
-		pub fn disable_trading_pair(
+		pub fn enable_trading_pair(
 			origin: OriginFor<T>,
 			currency_id_a: CurrencyId,
 			currency_id_b: CurrencyId,
 		) -> DispatchResultWithPostInfo {
 			T::ListingOrigin::ensure_origin(origin)?;
+
 			let trading_pair = TradingPair::from_token_currency_ids(currency_id_a, currency_id_b)
 				.ok_or(Error::<T>::InvalidCurrencyId)?;
 
 			match Self::trading_pair_statuses(trading_pair) {
-				// will disable Enabled trading_pair
-				TradingPairStatus::<_, _>::Enabled => {
-					TradingPairStatuses::<T>::insert(trading_pair, TradingPairStatus::NotEnabled);
-					Self::deposit_event(Event::DisableTradingPair(trading_pair));
+				TradingPairStatus::<_, _>::Disabled => {
+					TradingPairStatuses::<T>::insert(trading_pair, TradingPairStatus::Enabled);
+					Self::deposit_event(Event::EnableTradingPair(trading_pair));
 				}
-				// will disable Provisioning trading_pair
-				TradingPairStatus::<_, _>::Provisioning(_) => {
-					let module_account_id = Self::account_id();
-
-					// refund provision
-					for (who, contribution) in ProvisioningPool::<T>::drain_prefix(trading_pair) {
-						T::Currency::transfer(trading_pair.0, &module_account_id, &who, contribution.0)?;
-						T::Currency::transfer(trading_pair.1, &module_account_id, &who, contribution.1)?;
+				TradingPairStatus::<_, _>::Provisioning(provisioning_parameters) => {
+					let (total_provision_0, total_provision_1) = provisioning_parameters.accumulated_provision;
+
+					if total_provision_0.is_zero() && total_provision_1.is_zero() {
+						// if these's no provision, enable trading pair directly
+						TradingPairStatuses::<T>::insert(trading_pair, TradingPairStatus::Enabled);
+						Self::deposit_event(Event::EnableTradingPair(trading_pair));
+					} else if frame_system::Pallet::<T>::block_number() >= provisioning_parameters.not_before
+						&& (total_provision_0 >= provisioning_parameters.target_provision.0
```

Comment:
I think we should add another condition `total_provision_0 > 0 && total_provision_1 > 0`